md/single-sign-on-with-kerberos.md

```diff
@@ -0,0 +1,331 @@
+
+-	Client: the client is the end user’s machine. The client is referred to as the "Subject". It uses the credentials of an AD user account and is member of the domain.
+
+
+## Kerberos overview for Bonita
+
+The target architecture for integrating Kerberos SSO with Bonita and Spnego will involve a custom web application filter: Bonita Kerberos filter.
+
+![Authentication over Kerberos](images/kerberos-overview.png)<!--{.img-responsive}-->
+
+The scenario covered by the filter is the following: 
+
+- The Bonita Kerberos filter intercepts all the requests to bonita portal pages, and checks if the user is already logged in on Bonita 
+    
+    - If already logged in => Allow the access
+    - If not logged in => The request is transferred to the Spnego filter
+
+The Spnego filter will then verify the user’s Kerberos tickets if present or create a new one. After obtaining a valid ticket the filter will store some information (the authenticated user login) in the client request and get back to the Bonita Kerberos filter.
+	
+- The Bonita Kerberos filter will automatically create a Bonita session and let the user through to access the Portal resources.
+	
+:::info 
+**Note:** The Spnego filter is called programmatically by the Bonita Kerberos filter. It is not deployed as a standard web application filter in the application descriptor (web.xml). This is done in order to allow bi-directional exchanges with the Bonita Kerberos filter (regular web filters are only chained in one direction).
+:::
+
+::: warning  
+ Bonita "username" should match the authenticated user login returned in the client response. 
+:::
+
+## Pre-installation Environment Checks
+
+In this section, we will describe the minimal set of configuration checks that must be performed before starting the installation.
+
+### Domain Membership Checks
+
+Make sure that the Service Server AND the Client machines are member of the same AD Domain.
+
+### Service Principal Setup
+
+The Windows service that will be connected with SSO needs to be associated with a Principal in the form of an AD service account. This account is called the Service Principal Name (SPN).
+Note that this a different account than the machine object on which the service is running.
+
+1- Obtain from your AD administrator the AD service account that is going to be used as the SPN. If none exist yet, follow the steps below else, ignore this.
+
+2- Check that the service account is not already associated to any host by running this command (replace bonita.tomcat with your service account name):
+
+```
+setspn -L bonita.tomcat
+```
+This will list the hosts associated to this service account. There should not be any entry.
+
+3- Run these two commands in order to declare the SPN (bonita.tomcat). The first command will use the simple host name (win2008tomcat):
```

Comment:
is it always win2008tomcat ?

Reply:
no, it depends on what you use as hostname of your service machine.

Reply:
:+1: